### PR TITLE
Integration tests with capybara

### PR DESCRIPTION
**Pull request by why-a-en**
# Branch changes

- Fix N+1 query problem
Before:
Screenshot (73)
After:
Screenshot (73)

- Add integration tests for each page

### CODE REVIEW

File: spec/system/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  before do
+    driven_by(:rack_test)
+    @user1 = User.create!(name: 'anbehindY', photo: 'person.jpg', bio: 'Full-stack web developer')
+    @user2 = User.create!(name: 'picasso', photo: 'person2.jpg', bio: 'Full-stack web developer')
+    @post1 = Post.create!(author: @user1, title: 'First Post', text: 'This is my first post!')
+    @post2 = Post.create!(author: @user2, title: 'first post', text: 'This is my first post!')
+  end
+
+  context 'when the page is loaded' do
+    it 'shows the username of all users' do
```
Comment (brainconnect93) [inline spec/system/user_index_spec.rb:13]: - [x] Great effort working on your integration spec milestone, but currently i notice 2 of the test cases are not passing and requires your attention. Kindly work on them and be sure all the test cases are passing before you re-submit for review in order to get your project approve. 👍 👍 

Screenshot 👇 

image

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @anbehindY & @Lul-Abdifan, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- integration specs for views and fix n+1 problems ✔️
- Good use of Git flow ✔️

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/user_index_spec.rb (hunk 1 above)
Comment (why-a-en) [reply spec/system/user_index_spec.rb:13]: Thanks @brainconnect93. I fixed it already.

Comment (why-a-en) [reply spec/system/user_index_spec.rb:13]: The new commit is not showing even though I updated the code base and fix the problem. After researching the GitHub issue, it seems to be fixed automatically after some time. So I will just use this as a review request. I hope the new reviewer sees this comment.

**Review by uwadonat — APPROVED** [review]
## Hi @anbehindY 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Well organized and clean project ✔️
- No linter errors ✔️
- Correct workflow was used ✔️

appr

# Status: Approved ✔️✔️✔️


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**